webonary-cloud-api/lambda/__tests__/loadAuthorize.test.ts

```diff
@@ -0,0 +1,108 @@
+import { CustomAuthorizerEvent, Context } from 'aws-lambda';
+import axios from 'axios';
+import lambdaHandler from '../loadAuthorize';
+
+jest.mock('axios');
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+const username = 'testUser';
+const password = 'testPassword!';
+const base64Creds = Buffer.from(`${username}:${password}`).toString('base64');
+const headers = { Authorization: `Basic ${base64Creds}` };
+
+const dictionary = 'testDictionary';
+
+const event: CustomAuthorizerEvent = {
+  type: 'testEventType',
+  methodArn: 'POST/load/entry/',
+  pathParameters: { dictionary },
+  headers,
+};
+
+const context: Context = {
+  callbackWaitsForEmptyEventLoop: false,
+  functionName: '',
+  functionVersion: '',
+  invokedFunctionArn: '',
+  memoryLimitInMB: '',
+  awsRequestId: '',
+  logGroupName: '',
+  logStreamName: '',
+  getRemainingTimeInMillis: () => 0,
+  done: () => undefined,
+  fail:() => undefined,
+  succeed: () => undefined,
+};
+
+describe('loadAuthorize', () => {
+  test('succesful auth', async (): Promise<void> => {
+    mockedAxios.post.mockImplementation(() => Promise.resolve({ status: 200, data: '' }));
+    await lambdaHandler(event, context, (error, result) => {
+      expect(error).toBe(null);
+      expect(result.principalId).toEqual(`${dictionary}::${username}`);
+      expect(result.policyDocument.Statement[0].Effect).toBe('Allow');
+      expect(result.policyDocument.Statement[0].Action).toBe('execute-api:Invoke');
+      expect(result.policyDocument.Statement[0].Resource).toBe('POST/load/*/');
+    });
+
+    return expect.hasAssertions();
+  });
+
+  test('auth denied', async (): Promise<void> => {
+    mockedAxios.post.mockImplementation(() => Promise.resolve({ status: 200, data: 'some error' }));
```

Comment:
I would expect a status of 401.

Reply:
Webonary.org returns a 200 with an error message if the supplied user/password is not correct or is for a non-admin for the dictionary.

On the other hand, once AWS API gateway knows that a credential had been rejected, then it stores that info and issues 401 the next time it is used.

Reply:
👍